apps/console-lite/src/app/components/simple-market-list/simple-market-toolbar.spec.tsx

```diff
@@ -128,4 +136,64 @@ describe('SimpleMarketToolbar', () => {
       );
     });
   });
+
+  it('stateChange callback should work well', () => {
+    jest.spyOn(Router, 'useParams').mockReturnValue({
+      asset: 'asset1',
+      product: 'product1',
+      state: 'state1',
+    });
+    render(
+      <MockedProvider mocks={[]} addTypename={false}>
+        <SimpleMarketToolbar data={filterData as SimpleMarkets_markets[]} />
+      </MockedProvider>,
+      { wrapper: BrowserRouter }
+    );
+    fireEvent.click(screen.getByTestId('state-trigger'));
+    waitFor(() => {
+      expect(screen.getByRole('menu')).toBeInTheDocument();
+      fireEvent.click(screen.getByText('Suspended'));
+
+      expect(mockedNavigate).toHaveBeenCalledWith(
+        '/markets/STATE_SUSPENDED/product1/asset1'
+      );
+    });
+  });
+
+  it('stateChange callback should call navigate with url with state only', () => {
+    jest.spyOn(Router, 'useParams').mockReturnValue({});
+    render(
+      <MockedProvider mocks={[]} addTypename={false}>
+        <SimpleMarketToolbar data={filterData as SimpleMarkets_markets[]} />
+      </MockedProvider>,
+      { wrapper: BrowserRouter }
+    );
+    fireEvent.click(screen.getByTestId('state-trigger'));
+    waitFor(() => {
+      expect(screen.getByRole('menu')).toBeInTheDocument();
+      fireEvent.click(screen.getByText('Closed'));
+
+      expect(mockedNavigate).toHaveBeenCalledWith('/markets/STATE_CLOSED');
```

Comment:
question (non-blocking): and here, too, in case it's based on some constant, like the MarketState, might be better to use the enum value, not sure.

Reply:
yes, like above :+1:.
Meanwhile I found some other problems, and there are also some other changes..